pkg/apis/ceph.rook.io/v1/types.go

```diff
@@ -680,6 +680,17 @@ type PoolSpec struct {
 	// +optional
 	// +nullable
 	Quotas QuotaSpec `json:"quotas,omitempty"`
+
+	// The pg settings
+	// +optional
+	// +nullable
+	PGs int `json:"pgs,omitempty"`
+
+	// The autoscale mode settings
+	// +kubebuilder:validation:Enum=on;off;warn;""
+	// +optional
+	// +nullable
+	AutoscaleMode string `json:"autoscaleMode,omitempty"`
```

Comment:
Typically, for k8s APIs, the `string` should be a custom type like `type PoolAutoscaleMode string`, and then there should be values for those

```
const (
  PoolAutoscaleModeOn = "on"
  PoolAutoscaleModeOff = "off"
  PoolAutoscaleModeWarn = "warn"
)
```